src/main/scala/tech/sourced/gemini/FileQuery.scala

```diff
@@ -136,10 +130,21 @@ class FileQuery(conn: Session,
     result
   }
 
-  protected def hashFile(features: Iterable[Feature], docFreqFile: File): Array[Array[Long]] = {
-    log.info(s"Reading docFreq from ${docFreqFile.getAbsolutePath}")
-    val docFreq = OrderedDocFreq.fromJson(docFreqFile)
+  protected def readDocFreq(): Option[OrderedDocFreq] = {
+    log.info(s"Reading docFreq from DB")
+    val cols = tables.docFreqCols
+    val row = conn.execute(s"SELECT * FROM ${tables.docFreq} WHERE ${cols.id} = '1'").one()
+    if (row == null) {
+      None
+    } else {
+      val javaDf = row.getMap("df", classOf[java.lang.String], classOf[java.lang.Integer])
+      val df = javaDf.mapValues(Integer2int(_))
```

Comment:
Related to [comment on Java/Scala converters](https://github.com/src-d/gemini/pull/151/files#r200662238) - without changing the way DB is queried, it would be more natural to have something like

```scala
      val df = row
        .getMap("df", classOf[java.lang.String], classOf[java.lang.Integer])
        .asScala
        .mapValues(_.toInt)
```